From: Hannes Reinecke <hare@suse.de>
Date: Wed, 2 Oct 2013 09:05:05 +0200
Subject: scsi_dh_alua: simplify alua_check_sense()
References: bnc#843642
Patch-Mainline: not yet

alua_check_sense() only need to specify those cases which are
different from the original handling.

Signed-off-by: Hannes Reinecke <hare@suse.de>
---
 drivers/scsi/device_handler/scsi_dh_alua.c | 15 ---------------
 1 file changed, 15 deletions(-)

diff --git a/drivers/scsi/device_handler/scsi_dh_alua.c b/drivers/scsi/device_handler/scsi_dh_alua.c
index bab2587..765ed25 100644
--- a/drivers/scsi/device_handler/scsi_dh_alua.c
+++ b/drivers/scsi/device_handler/scsi_dh_alua.c
@@ -657,21 +657,6 @@ static int alua_check_sense(struct scsi_device *sdev,
 			alua_check(sdev);
 			return ADD_TO_MLQUEUE;
 		}
-		if (sense_hdr->asc == 0x04 && sense_hdr->ascq == 0x0b)
-			/*
-			 * LUN Not Accessible -- Target port in standby state
-			 */
-			return SUCCESS;
-		if (sense_hdr->asc == 0x04 && sense_hdr->ascq == 0x0c)
-			/*
-			 * LUN Not Accessible -- Target port in unavailable state
-			 */
-			return SUCCESS;
-		if (sense_hdr->asc == 0x04 && sense_hdr->ascq == 0x12)
-			/*
-			 * LUN Not Ready -- Offline
-			 */
-			return SUCCESS;
 		break;
 	case UNIT_ATTENTION:
 		if (sense_hdr->asc == 0x29 && sense_hdr->ascq == 0x00)
-- 
1.7.12.4

